server/apps/publish/published_item.py

```diff
@@ -278,15 +278,15 @@ def find_one(self, req, **lookup):
         return item
 
     def __set_published_item_expiry(self, doc):
+        """
+        Set the expiry for the published item
+        :param dict doc: doc on which publishing action is performed
+        """
         desk_id = doc.get('task', {}).get('desk', None)
-        desk = {}
-
-        if desk_id:
-            desk = get_resource_service('desks').find_one(req=None, _id=desk_id)
+        stage_id = doc.get('task', {}).get('stage', None)
 
-        expiry_minutes = desk.get('published_item_expiry', config.PUBLISHED_ITEMS_EXPIRY_MINUTES)
-        doc['expiry'] = get_expiry_date(expiry_minutes, offset=doc[EMBARGO]) if doc.get(EMBARGO) else \
-            get_expiry_date(expiry_minutes)
+        doc['expiry'] = get_expiry(desk_id, stage_id, offset=doc[EMBARGO]) if doc.get(EMBARGO) else \
```

Comment:
it could be just `get_expiry(..,.., offset=doc.get(EMBARGO))` given that it handles `offset=None` situation